From: Sang-Heon Jeon <ekffu200098@gmail.com>
Subject: riscv: remove unused __late_set_fixmap() and __late_clear_fixmap()
Date: Thu, 9 Jul 2026 02:06:46 +0900

__late_set_fixmap() and __late_clear_fixmap() are only used after
early_ioremap_reset() has been called.  riscv never calls it because
__set_fixmap() works before and after paging_init().  So remove them.

No functional change.

Link: https://lore.kernel.org/20260708170647.362562-3-ekffu200098@gmail.com
Signed-off-by: Sang-Heon Jeon <ekffu200098@gmail.com>
Cc: Albert Ou <aou@eecs.berkeley.edu>
Cc: Alexandre Ghiti <alex@ghiti.fr>
Cc: Catalin Marinas <catalin.marinas@arm.com>
Cc: David Hildenbrand <david@kernel.org>
Cc: Liam R. Howlett <liam@infradead.org>
Cc: Lorenzo Stoakes <ljs@kernel.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Palmer Dabbelt <palmer@dabbelt.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@kernel.org>
Cc: Will Deacon <will@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/riscv/include/asm/fixmap.h |    3 ---
 1 file changed, 3 deletions(-)

--- a/arch/riscv/include/asm/fixmap.h~riscv-remove-unused-__late_set_fixmap-and-__late_clear_fixmap
+++ a/arch/riscv/include/asm/fixmap.h
@@ -55,9 +55,6 @@ enum fixed_addresses {
 
 #define __early_set_fixmap	__set_fixmap
 
-#define __late_set_fixmap	__set_fixmap
-#define __late_clear_fixmap(idx) __set_fixmap((idx), 0, FIXMAP_PAGE_CLEAR)
-
 extern void __set_fixmap(enum fixed_addresses idx,
 			 phys_addr_t phys, pgprot_t prot);
 
_
